validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorIndexProvider.java

```diff
@@ -99,6 +100,13 @@ private void logNewValidatorIndices(final Map<BLSPublicKey, Integer> knownValida
     }
   }
 
+  public Optional<BLSPublicKey> getPublicKey(final Integer index) {
+    return validatorIndicesByPublicKey.entrySet().stream()
+        .filter(entry -> entry.getValue().equals(index))
+        .findFirst()
+        .map(Map.Entry::getKey);
+  }
+
```

Comment:
looks like a perfect unit test opportunity to be honest

Reply:
Added a unit test